helper/keystore/keystore.go

```diff
@@ -9,32 +9,39 @@ import (
 
 type createFn func() ([]byte, error)
 
-type readFn func(b []byte) (interface{}, error)
-
-func CreateIfNotExists(path string, create createFn, read readFn) (interface{}, error) {
+// CreateIfNotExists generates a private key at the specified path,
+// or reads it if a key file is present
+func CreateIfNotExists(path string, create createFn) ([]byte, error) {
 	_, err := os.Stat(path)
 	if err != nil && !os.IsNotExist(err) {
 		return nil, fmt.Errorf("failed to stat (%s): %v", path, err)
 	}
-	if os.IsNotExist(err) {
-		// create the file
-		buf, err := create()
+
+	var keyBuff []byte
+	if !os.IsNotExist(err) {
+		// Key exists
+
+		// Read the Base64 encoded key from the disk.
+		// Base64 decoding is done later on
+		keyBuff, err = ioutil.ReadFile(path)
```

Comment:
I would remove this comment, since it lies a bit 🙂 
it reads **anything** on the `path`. Doesn't care if it's a key or not.
```suggestion
		keyBuff, err = ioutil.ReadFile(path)
```

Reply:
Fixed 👍